### PR TITLE
Allow passing `buildScriptName` as addon option

### PR DESCRIPTION
**Pull request by tmeasday**
**To QA**

1. Rename the `build-storybook` script in `package.json` to something else
2. Run `yst` and try to start a build. You should see an error in the terminal
3. Update [`main.js`'s prod](https://github.com/chromaui/addon-visual-tests/blob/45ed61b61d1d98e86c47cc4067bf4b297b404f1d/.storybook/main.ts#L4-L8) settings to include `buildScriptName: 'name-you-chose'`
4. Try step 2 again.

We should add some documentation about this @jonniebigodes.

I'm not sure exactly how we can give a better error (right now it talks about the CLI and the `--build-script-name` option). Probably we need to think more about logs + errors coming from the CLI -> addon as part of showing build progress more generally, so I think we ship this for now and iterate.

<details>
  <summary>📦 Published PR as canary version: <code>0.0.37--canary.44.4957b7e.0</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @chromaui/addon-visual-tests@0.0.37--canary.44.4957b7e.0
  # or 
  yarn add @chromaui/addon-visual-tests@0.0.37--canary.44.4957b7e.0
  ```
</details>

### CODE REVIEW

**Comment by jonniebigodes** [comment]
@tmeasday, thanks for the callout. I will update the internal user guide to feature this in the meantime.

**Comment by tmeasday** [comment]
@jonniebigodes sure thing, every PR is canaried. I'm not sure why it didn't show up as a comment on the PR but here it is `@chromaui/addon-visual-tests@0.0.32--canary.45ed61b.0` (you can get it from the [`Release` GH action](https://github.com/chromaui/addon-visual-tests/actions/runs/5946946375/job/16128280826?pr=44))

**Review by yannbf — APPROVED** [review]
Hey @tmeasday I have tested this out in the storybook monorepo and it worked great!

File: src/index.ts [hunk 1]
```diff
@@ -46,22 +46,29 @@ const observeGitInfo = async (
 
 async function serverChannel(
   channel: Channel,
-  { projectToken: initialProjectToken }: { projectToken: string }
+  {
+    projectToken: initialProjectToken,
+    buildScriptName,
+  }: { projectToken?: string; buildScriptName?: string }
 ) {
+  console.log({ initialProjectToken, buildScriptName });
```
Comment (jonniebigodes) [inline src/index.ts:54]: @tmeasday, before we forget, could you remove this line so we don't accidentally expose the information in the CLI? I'm assuming that this was added for testing purposes, correct?

**Comment by jonniebigodes** [comment]
@tmeasday same here. All other pull requests got the canary version mentioned, but this didn't, probably GitHub being GitHub 😄 , hence why I asked. But nonetheless appreciate the follow-up. Next time this happens, I know where to look. I've tested it out, and it's worked great. Left one minor comment, but this should be good to go.

File: src/index.ts [hunk 2]
```diff
@@ -74,7 +81,8 @@ async function serverChannel(
             started = true;
           }
         },
-      } as any,
+        // as any due to CLI mistyping: https://github.com/chromaui/chromatic-cli/pull/800
+      } as Partial<Parameters<typeof run>[0]["options"]> as any,
```
Comment (ghengeveld) [inline src/index.ts:85]: I think this can be updated now that the fix is released in `chromatic@6.23.1`.

**Review by ghengeveld — APPROVED** [review]

File: src/index.ts (hunk 1 above)
Comment (tmeasday) [reply src/index.ts:54]:
```suggestion
```

Comment (tmeasday) [reply src/index.ts]: Good pickup, thanks @jonniebigodes

File: src/index.ts (hunk 2 above)
Comment (tmeasday) [reply src/index.ts:84]: Will do after merging

**Comment by work933k** [comment]
I'm working with NX. Having to use "buildScriptName" is inconvenient. Would it be possible to also add an option to define "buildScript". This so: `npx nx run shared-material-theme-dummy:build-storybook` would be possible.

**Pull request merged**